spring-cloud-task-core/src/test/java/org/springframework/cloud/task/repository/dao/JdbcTaskExecutionDaoTests.java

```diff
@@ -166,6 +174,120 @@ public void testFindAllDefaultSort()  {
+		assertEquals(22, dateTimeFoo1.getDayOfMonth());
+		assertEquals(23, dateTimeFoo1.getHour());
+		assertEquals(59, dateTimeFoo1.getMinute());
+		assertEquals(0, dateTimeFoo1.getSecond());
+
+		final LocalDateTime dateTimeFoo3 = LocalDateTime.ofInstant(latestTaskExecutions.get(1).getStartTime().toInstant(), ZoneOffset.UTC);
+
+		assertEquals(2016, dateTimeFoo3.getYear());
+		assertEquals(Month.AUGUST, dateTimeFoo3.getMonth());
+		assertEquals(20, dateTimeFoo3.getDayOfMonth());
+		assertEquals(14, dateTimeFoo3.getHour());
+		assertEquals(45, dateTimeFoo3.getMinute());
+		assertEquals(0, dateTimeFoo3.getSecond());
+
+		final LocalDateTime dateTimeFoo4 = LocalDateTime.ofInstant(latestTaskExecutions.get(2).getStartTime().toInstant(), ZoneOffset.UTC);
+
+		assertEquals(2015, dateTimeFoo4.getYear());
+		assertEquals(Month.FEBRUARY, dateTimeFoo4.getMonth());
+		assertEquals(20, dateTimeFoo4.getDayOfMonth());
+		assertEquals(14, dateTimeFoo4.getHour());
+		assertEquals(45, dateTimeFoo4.getMinute());
+		assertEquals(0, dateTimeFoo4.getSecond());
+	}
+
+	@Test
+	@DirtiesContext
+	public void testFindLatestTaskExecutions3()  {
+		initializeRepositoryNotInOrderWithMultipleTaskExecutions();
+		final List<TaskExecution> latestTaskExecutions = dao.getLatestTaskExecutionsByTaskNames("FOO5");
+		assertTrue("Expected only 1 taskExecution but got " + latestTaskExecutions.size(), latestTaskExecutions.size() == 1);
+		final LocalDateTime dateTime = LocalDateTime.ofInstant(latestTaskExecutions.get(0).getStartTime().toInstant(), ZoneOffset.UTC);
+
+		assertEquals(2015, dateTime.getYear());
+		assertEquals(Month.FEBRUARY, dateTime.getMonth());
+		assertEquals(22, dateTime.getDayOfMonth());
+		assertEquals(23, dateTime.getHour());
+		assertEquals(59, dateTime.getMinute());
+		assertEquals(0, dateTime.getSecond());
+	}
+
+	@Test
+	@DirtiesContext
+	public void testGetLatestTaskExecutionForTaskName1()  {
+		initializeRepositoryNotInOrderWithMultipleTaskExecutions();
+		final TaskExecution latestTaskExecution = dao.getLatestTaskExecutionForTaskName("Bar5");
+		assertNull("Expected the latestTaskExecution to be null but got" + latestTaskExecution, latestTaskExecution);
+	}
+
+	@Test
+	@DirtiesContext
+	public void testGetLatestTaskExecutionForTaskName2()  {
+		initializeRepositoryNotInOrderWithMultipleTaskExecutions();
+		final TaskExecution latestTaskExecution = dao.getLatestTaskExecutionForTaskName("FOO5");
+		assertNotNull("Expected the latestTaskExecution not to be null", latestTaskExecution);
+
+		final LocalDateTime dateTime = LocalDateTime.ofInstant(latestTaskExecution.getStartTime().toInstant(), ZoneOffset.UTC);
+
+		assertEquals(2015, dateTime.getYear());
+		assertEquals(Month.FEBRUARY, dateTime.getMonth());
+		assertEquals(22, dateTime.getDayOfMonth());
+		assertEquals(23, dateTime.getHour());
+		assertEquals(59, dateTime.getMinute());
+		assertEquals(0, dateTime.getSecond());
+	}
+
+	@Test
+	@DirtiesContext
+	public void testGetLatestTaskExecutionForTaskName3()  {
```

Comment:
Couldn't you add this test to testGetLatestTaskExecutionForTaskName2?   Since it is testing the same premise but with a different criteria?